src/util/utils.js

```diff
@@ -218,6 +218,47 @@ const extractStackTraceUptoLastSubstringMatch = (trace, stringLiterals) => {
   return traceLines.slice(0, lastRelevantIndex + 1).join('\n');
 };
 
+/**
+ * Gets the current value of the USE_HAS_DYNAMIC_CONFIG_FLAG environment variable.
+ * This is a getter function to allow for easier testing by reading the env var each time.
+ *
+ * @returns true if the hasDynamicConfig flag should be used, false for legacy behavior
+ */
```

Comment:
what happens in case of process.env.USE_HAS_DYNAMIC_CONFIG_FLAG is undefined ? better to handle it too - otherwise, environment variable need to be defined even before the release

Reply:
by default it is enabled but if we want to disable it them we need to set false what behavior do we want?

Reply:
to disable by default & enable in staggered way by close monitoring

Reply:
https://github.com/rudderlabs/rudderstack-operator/pull/3714 released operator pr